included/7.0.1/Dockerfile

```diff
@@ -5,65 +5,48 @@
 # build this image with command
 #   docker build -t cypress/included:7.0.1 .
 #
-FROM cypress/browsers:node14.16.0-chrome89-ff77
-
-# Update the dependencies to get the latest and greatest (and safest!) packages.
-RUN apt update && apt upgrade -y
+FROM cypress/browsers:node14.16.0-chrome89-ff86
 
 # avoid too many progress messages
 # https://github.com/cypress-io/cypress/issues/1243
-ENV CI=1
-
-# disable shared memory X11 affecting Cypress v4 and Chrome
-# https://github.com/cypress-io/cypress-docker-images/issues/270
-ENV QT_X11_NO_MITSHM=1
-ENV _X11_NO_MITSHM=1
-ENV _MITSHM=0
+ENV CI=1 \
+  # disable shared memory X11 affecting Cypress v4 and Chrome
+  # https://github.com/cypress-io/cypress-docker-images/issues/270
+  QT_X11_NO_MITSHM=1 \
+  _X11_NO_MITSHM=1 \
+  _MITSHM=0 \
+  # point Cypress at the /root/cache no matter what user account is used
+  # see https://on.cypress.io/caching
+  CYPRESS_CACHE_FOLDER=/root/.cache/Cypress
 
 # should be root user
-RUN echo "whoami: $(whoami)"
-RUN npm config -g set user $(whoami)
-
-# command "id" should print:
-# uid=0(root) gid=0(root) groups=0(root)
-# which means the current user is root
-RUN id
-
-# point Cypress at the /root/cache no matter what user account is used
-# see https://on.cypress.io/caching
-ENV CYPRESS_CACHE_FOLDER=/root/.cache/Cypress
-RUN npm install -g "cypress@7.0.1"
-RUN cypress verify
-
-# Cypress cache and installed version
-# should be in the root user's home folder
-RUN cypress cache path
-RUN cypress cache list
-RUN cypress info
-RUN cypress version
-
-# give every user read access to the "/root" folder where the binary is cached
-# we really only need to worry about the top folder, fortunately
-RUN ls -la /root
-RUN chmod 755 /root
-
-# always grab the latest Yarn
-# otherwise the base image might have old versions
-# NPM does not need to be installed as it is already included with Node.
-RUN npm i -g yarn@latest
-
-# Show where Node loads required modules from
-RUN node -p 'module.paths'
-
-# should print Cypress version
-# plus Electron and bundled Node versions
-RUN cypress version
-RUN echo  " node version:    $(node -v) \n" \
-  "npm version:     $(npm -v) \n" \
-  "yarn version:    $(yarn -v) \n" \
-  "debian version:  $(cat /etc/debian_version) \n" \
-  "user:            $(whoami) \n" \
-  "chrome:          $(google-chrome --version || true) \n" \
-  "firefox:         $(firefox --version || true) \n"
+RUN echo "whoami: $(whoami)" \
+  && npm config -g set user $(whoami) \
+  # command "id" should print:
+  # uid=0(root) gid=0(root) groups=0(root)
+  # which means the current user is root
+  && id \
+  && npm install -g "cypress@7.0.1" --force \
```

Comment:
Is cypress itself **REALLY** 650Mb?! O_o

Reply:
Does `-g` automatically exclude dev dependencies? Have you tried if adding `--only=prod` makes a difference?

Just an idea. Over 600 MB really is massive.

Reply:
Will give it a shot - 650Mb for one dependency is indeed gigantic.

Reply:
`--only=prod` has no effect: the big offender here is the post-install hook for cypress:

```
> cypress@7.0.1 postinstall /usr/local/lib/node_modules/cypress
> node index.js --exec install

[19:07:20]  Downloading Cypress     [started]
[19:07:31]  Downloading Cypress     [completed]
[19:07:31]  Unzipping Cypress       [started]
[19:07:41]  Unzipping Cypress       [completed]
[19:07:41]  Finishing Installation  [started]
[19:07:41]  Finishing Installation  [completed]
```

Reply:
Yeah, that requires fixing the problem in upstream cypress: the size of cypress itself (the download in the above comment) is absolutely ridiculous, even for individual components:

```
root@024f6d63aba0:~# du -hs /root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/*
9.5M	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/desktop-gui
56K	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/driver
39M	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/electron
1.3M	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/example
6.4M	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/extension
8.0M	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/https-proxy
11M	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/launcher
6.9M	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/net-stubbing
17M	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/network
8.2M	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/proxy
24K	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/reporter
11M	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/rewriter
20K	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/root
12M	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/runner
118M	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/runner-ct
345M	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/server
11M	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/server-ct
4.8M	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/socket
160K	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/static
112K	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/ts
24K	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/ui-components
16K	/root/.cache/Cypress/7.0.1/Cypress/resources/app/packages/web-config
```

Reply:
Sorry, that was just a quick idea. Doesn't cypress come with a bundled electron? 🤔
Maybe the maintainers can shed some light.
Thanks for your improvements so far. 👍

Reply:
I tried looking into bundling cypress via `git clone`, but overall I cannot understand any of their build processes: it's too complex/entangled for my expertise, and a simple `npm install` already fails for me.